nexus/src/external_api/http_entrypoints.rs

```diff
@@ -1414,6 +1394,128 @@ async fn ip_pool_update(
+
+        let path = path_params.into_inner();
+        let pool_lookup = nexus.ip_pool_lookup(&opctx, &path.pool)?;
+
+        let assocs = nexus
+            .ip_pool_silo_list(&opctx, &pool_lookup, &pag_params)
+            .await?
+            .into_iter()
+            .map(|assoc| assoc.into())
+            .collect();
+
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            assocs,
+            &|_, x: &views::IpPoolSilo| x.silo_id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Make an IP pool available within a silo
+#[endpoint {
+    method = POST,
+    path = "/v1/system/ip-pools/{pool}/silos",
+    tags = ["system/networking"],
+}]
+async fn ip_pool_silo_link(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    path_params: Path<params::IpPoolPath>,
+    resource_assoc: TypedBody<params::IpPoolSiloLink>,
+) -> Result<HttpResponseCreated<views::IpPoolSilo>, HttpError> {
+    let apictx = rqctx.context();
+    let handler = async {
+        let opctx = crate::context::op_context_for_external_api(&rqctx).await?;
+        let nexus = &apictx.nexus;
+        let path = path_params.into_inner();
+        let resource_assoc = resource_assoc.into_inner();
+        let pool_lookup = nexus.ip_pool_lookup(&opctx, &path.pool)?;
+        let assoc = nexus
+            .ip_pool_link_silo(&opctx, &pool_lookup, &resource_assoc)
+            .await?;
+        Ok(HttpResponseCreated(assoc.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Unlink an IP pool from a silo
+///
+/// Will fail if there are any outstanding IPs allocated in the silo.
+#[endpoint {
+    method = DELETE,
+    path = "/v1/system/ip-pools/{pool}/silos/{silo}",
+    tags = ["system/networking"],
+}]
+async fn ip_pool_silo_unlink(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    path_params: Path<params::IpPoolSiloPath>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let apictx = rqctx.context();
+    let handler = async {
+        let opctx = crate::context::op_context_for_external_api(&rqctx).await?;
+        let nexus = &apictx.nexus;
+        let path = path_params.into_inner();
+        let pool_lookup = nexus.ip_pool_lookup(&opctx, &path.pool)?;
+        let silo_lookup = nexus.silo_lookup(&opctx, path.silo)?;
+        nexus.ip_pool_unlink_silo(&opctx, &pool_lookup, &silo_lookup).await?;
+        Ok(HttpResponseUpdatedNoContent())
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Make an IP pool default or not-default for a silo
+///
+/// When a pool is made default for a silo, any existing default will remain
+/// linked to the silo, but will no longer be the default.
+#[endpoint {
+    method = PUT,
+    path = "/v1/system/ip-pools/{pool}/silos/{silo}",
+    tags = ["system/networking"],
+}]
+async fn ip_pool_silo_update(
```

Comment:
Yeah, it could be `ip_pool_silo_set_default`. There are no other properties of the link that can be updated. Another option would be `ip_pool_silo_link_update` to make clearer it's about updating the link and not the pool or the silo.